src/circuit/gadget/ecc/chip/mul_fixed.rs

```diff
@@ -0,0 +1,529 @@
+use std::array;
+
+use super::{
+    add, add_incomplete, copy, CellValue, EccConfig, EccPoint, EccScalarFixed, EccScalarFixedShort,
+    Var,
+};
+use crate::constants::{
+    self,
+    load::{OrchardFixedBase, OrchardFixedBasesFull, ValueCommitV, WindowUs},
+};
+
+use group::Curve;
+use halo2::{
+    circuit::Region,
+    plonk::{Advice, Column, ConstraintSystem, Error, Expression, Fixed, Permutation, Selector},
+    poly::Rotation,
+};
+use pasta_curves::{
+    arithmetic::{CurveAffine, FieldExt},
+    pallas,
+};
+
+pub mod full_width;
+pub mod short;
+
+// A sum type for both full-width and short bases. This enables us to use the
+// shared functionality of full-width and short fixed-base scalar multiplication.
+#[derive(Copy, Clone, Debug)]
+enum OrchardFixedBases {
+    Full(OrchardFixedBasesFull),
+    ValueCommitV,
+}
+
+impl From<OrchardFixedBasesFull> for OrchardFixedBases {
+    fn from(full_width_base: OrchardFixedBasesFull) -> Self {
+        Self::Full(full_width_base)
+    }
+}
+
+impl From<ValueCommitV> for OrchardFixedBases {
+    fn from(_value_commit_v: ValueCommitV) -> Self {
+        Self::ValueCommitV
+    }
+}
+
+impl OrchardFixedBases {
+    pub fn generator(self) -> pallas::Affine {
+        match self {
+            Self::ValueCommitV => constants::value_commit_v::generator(),
+            Self::Full(base) => {
+                let base: OrchardFixedBase = base.into();
+                base.generator
```

Comment:
The `base.into()` here computes the lagrange coefficients inline:
https://github.com/zcash/orchard/blob/37b1b7f357cd34d93f9c55bb96efd05ab4e84408/src/constants/load.rs#L24-L61
But we then immediately discard them and just return the generator itself. Instead, add a `generator()` method to `OrchardFixedBasesFull` which can sidestep deriving the entire `OrchardFixedBase`.